File: TimePlanner.Domain/Core/WorkItemsTracking/Status.cs
using TimePlanner.Domain.Core.WorkItemsTracking.Segments;
using TimePlanner.Domain.Core.WorkItemsTracking.WorkItems;

namespace TimePlanner.Domain.Core.WorkItemsTracking
{
  public record struct Status(
    DateTime StartedAt,
    TimeSpanValue Deposit,
    TimeSpanValue Pause,
    WorkingTime WorkingTime,
    List<WorkItem> WorkItems);

  public record struct WorkingTime(TimeSpanValue Distributed, TimeSpanValue Undistributed)
  {
    public TimeSpanValue Total => Distributed + Undistributed;
  }
}

File: frontend/time-planner/src/app/time-tracking/models/status.ts
import { WorkItem } from "./work-item";

export interface WorkingTime{
  distributed: string,
  undistributed: string,
  total: string
}
export interface Status {
  startedAt: string;
  breakStartedAt: string,
  deposit: string;
  pause: string;
  workingTime: WorkingTime,
  workItems: Array<WorkItem>
}